messageforums-hbm/src/java/org/sakaiproject/component/app/messageforums/dao/hibernate/MessageImpl.hbm.xml

```diff
@@ -808,6 +808,17 @@
              where area.contextId = :contextId and
              message.deleted = false]]>
   </query>
+
+    <!-- returns recent forum  threads for a given siteId -->
+    <query name="findRecentDiscussionForumThreadsInSite">
+        <![CDATA[select message, topic, forum from org.sakaiproject.component.app.messageforums.dao.hibernate.MessageImpl as message
```

Comment:
Performance on the query looks good. Joins have indexes, as does the inReplyTo column. The deleted column doesn't but an index wouldn't help much as most messages won't be deleted and it's a boolean.